scripts/generate-contract-addresses-doc-page.py

```diff
@@ -29,22 +30,40 @@ def main() -> None:
         )
     )
 
-    for chain_id, contracts in deployment["chains"].items():
-        name = _NAMES[int(chain_id)]
-        explorer = _EXPLORERS[int(chain_id)]
-        if name == "Ethereum":
-            base_contracts = deployment["base_chain"]
-            contracts.update(base_contracts)
+    contract_info: dict[ChainId, dict[str, ChecksumAddress]] = {}
+    for artifact_path in artifacts_dir.glob("*.deployment.json"):
+        deployment = Deployment.from_file(artifact_path)
+        base_chain_id = deployment.base.chain_id
+        if base_chain_id not in contract_info:
+            contract_info[base_chain_id] = {}
+        file_chain_id = str(base_chain_id)
+        if not artifact_path.stem == "base.deployment":
+            file_chain_id, _ = artifact_path.stem.split(".")[0].split("-")
```

Comment:
We should rather use information from the `deployment` instance, instead of parsing this from the filename.

Reply:
Also, note that you can distinguish between a base deployment and regular chain deployment by checking
`deployment.chain is not None` (for base deployment, it will be `None`).